spark/src/main/scala/ai/chronon/spark/Join.scala

```diff
@@ -317,13 +300,41 @@ class Join(joinConf: api.Join,
                     s"Macro ${Constants.ChrononRunDs} is only supported for single day join, current range is ${leftRange}")
                 }
 
+                val runSmallMode = {
+                  if (tableUtils.smallModelEnabled) {
+                    val thresholdCount: Int = unfilledLeftDf.map(_.count.toInt).getOrElse(0)
+                    val result = thresholdCount <= tableUtils.smallModeNumRowsCutoff
+                    if (result) {
+                      logger.info(s"Counted $thresholdCount rows, running join in small mode.")
+                    } else {
+                      logger.info(
+                        s"Counted greater than ${tableUtils.smallModeNumRowsCutoff} rows, proceeding with normal computation.")
+                    }
+                    result
+                  } else {
+                    false
+                  }
+                }
+
+                // create filter for leftDf
                 val bloomFilterOpt = if (runSmallMode) {
                   // If left DF is small, hardcode the key filter into the joinPart's GroupBy's where clause.
-                  injectKeyFilter(leftDf, joinPart)
+                  injectKeyFilter(unfilledLeftDf.map(_.df).get, joinPart)
```

Comment:
nit: can we move this line 322 to after line 308? 

`injectKeyFilter` is a function with side effect (it modifies joinPart in place) so it's risky and should be made obvious. right now it's part of computing `bloomFilterOpt` which is confusing. Let's refactor this part.